lib/src/rules/do_not_use_environment.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Do not use environment variables.';
```

Comment:
One small nit, which you can choose to ignore: the term "environment variable" usually refers to variables inherited by a process from the parent process. In Dart, those are accessed using `Platform.environment`. In the analyzer code base we refer to the `fromEnvironment` keys as "declared variables", but that term isn't used anywhere else that I'm aware of. I don't know of a good alternative, but "environment variable" might be confusing.

Reply:
I'm leaning towards "environment declared variables".

Reply:
Thanks!